plasTeX/Packages/tikz.py

```diff
@@ -0,0 +1,133 @@
+"""
+Implement the tikz package for html output.
+The original tikz and latex do most of the work and then pdf2svg (or similar
+software) turns it into a svg image.
+Needs Beautiful Soup, Jinja2, and pdf2svg or similar
+"""
+
+import os
+import shutil
+import tempfile
+from plasTeX import Environment, NoCharSubEnvironment, Macro
+from plasTeX.PackageResource import PackageResource
+
+from plasTeX.Logging import getLogger
+log = getLogger()
+
+try:
+    from jinja2 import Template
+except ImportError:
+    log.warning('Cannot find jinja2 lib. Cannot use tikz.')
+
+try:
+    from bs4 import BeautifulSoup
+except ImportError:
+    log.warning('Cannot find BeautifulSoup lib. Cannot use tikz.')
+
+
+class tikzpicture(NoCharSubEnvironment):
+    """
+    A tikz picture whose content will be converted in the processFileContent callback.
+    """
+    class matrix(Environment):
+        """
+        Avoids conflict with amsmath matrix thanks to the context stack
+        mechanism.
+        """
+
+    class draw(Macro):
+        """ Only avoids unrecognized command warning. """
+    
+    class fill(Macro):
+        """ Only avoids unrecognized command warning. """
+
+    class filldraw(Macro):
+        """ Only avoids unrecognized command warning. """
+
+    class node(Macro):
+        """ Only avoids unrecognized command warning. """
+
+    class path(Macro):
+        """ Only avoids unrecognized command warning. """
+
+    class clip(Macro):
+        """ Only avoids unrecognized command warning. """
+
+def tikzConvert(document, content, envname, placeholder):
+    tmp_dir = document.userdata[envname]['tmp_dir']
+    working_dir = document.userdata['working-dir']
+    target_dir = os.path.join(working_dir, document.config['files']['directory'], 'images')
+    template = document.userdata[envname]['template']
+    compiler = document.userdata[envname]['compiler']
+    pdf2svg = document.userdata[envname]['pdf2svg']
+
+    cwd = os.getcwd()
+    os.chdir(tmp_dir)
+    soup = BeautifulSoup(content, "html.parser")
+    encoding = soup.original_encoding
+
+    envs = soup.findAll(envname)
+    for env in envs:
+        object_id = env.attrs['id']
+        basepath = os.path.join(tmp_dir, object_id)
+        texpath = basepath + '.tex'
+        pdfpath = basepath + '.pdf'
+        svgpath =  basepath + '.svg'
+
+        context = {'tikzpicture': env.text.strip() }
+        template.stream(**context).dump(texpath, encoding)
+
+        os.system(compiler + ' ' + texpath)
```

Comment:
Can these os.system calls be replaced with subprocess.call?  If the paths contain spaces, this could fail.